### DISCUSSION TITLE
Support for SI4713 FM Transmitter with I2S input

### DISCUSSION THREAD

**Discussion by amstereo4ever** [schreibfaul1/ESP32-audioI2S#795] [Ideas]
Hello Schreibfaul/Wolle and community,

I am more of a hardware developer and I design devices to support old radios and TVs by supplying vintage signals. 

Currently there is a trend to switch off FM (UKW) broadcasting. Switzerland will do this in 2024. Germany will follow soon and other countries have similar plans. This will render many current radios useless. Luckily there are interesting ICs available that can generate an FM stereo signal of high quality. The Skyworks SI4713 is such a device. Unlike other such ICs it has I2S input capability which would make it an ideal companion for your library and the ESP32. 

There is a library for this chip made by Adafruit and they have designed a break out board. This board and library unfortunately only supports analog inputs and the input pins of the SI4713 for the I2S signal are not available. 

I have designed a new break our board that is more universal and gives access to the I2S pins as well as the analog inputs and the I2C  bus needed for control. The SI4713 supports RDS so it could even display song and artist information of played content if the library would relay this information to the SI4713 via I2C.

As my programming skills are relatively limited I am wondering if support for the FM transmitter could be implemeted in the library. For tests I could contrbute working break out boards. I have ordered those at JLCPCB and they should arrive in a few days.

I have read that some users have asked for BT implementation to forward the signals to BT devices. Due to the design of the ESP32 hardware this is not possible when WiFi is being used as the RF section is shared by WiFi and BT. Supporting the SI4713 could be an alternative way to distribute the signal played by the ESP32 to existing analog radios and stereo systems.

Just an idea that could make the library even more universal. I would contribute hardware and testing environment if needed.

Cheers

Semir

**Comment by al-davis** [comment]
I am interested in your Si4713 / I2S project using the Adafruit dev bd (available on Digikey.  I see no date with year called out.  Is this project still active  ?

**Comment by schreibfaul1** [comment]
It looks as if the Si4713 is an I2C controlled FM transmitter. You can connect this to the output of the DAC with an analogue audio cable. However, you will need additional pins SDA, SCL and CS. Perhaps there is already a library on the net that does the control, otherwise you will have to rely on your own development. It is easier to use an FM transmitter without I2C.

**Comment by al-davis** [comment]
I already have a project that receives internet stations, converts the
output to analog stereo and sends that to an external FM transmitter.
Based  on, I think your, project from the Hack-a-day website.

 I'd like to incorporate the FM transmit section into my project instead
of using the external transmitter.

I'm looking into using the Adafruit breakout board that uses the SI4713
chip and altering it to use the I2S inputs of the SI4713 chip.   You said
that you designed a new break out board that is more universal and gives
access to the I2S pins as well as the analog inputs and the I2C bus.  I'd
be interested in that board or any info that you have on that path.  I can
probably handle software to bring it all together.

Al

**Reply by schreibfaul1** [reply]
Are you sure there is an I2S input. It looks like I2C.

**Comment by al-davis** [comment]
According to this spec :
https://www.alldatasheet.com/html-pdf/792705/SILABS/SI4713/215/1/SI4713.html
   the SI4713 chip supports I2S (and I2c), the Adafruit board that uses
that chip 
https://learn.adafruit.com/adafruit-si4713-fm-radio-transmitter-with-rds-rdbs-support/pinouts
  does not bring I2S pins out.  Because of the chip package used on the
Adafruit board I don't know if the pins are accessible and I am not aware
of any other board using the SI4713 that is why your comment about another
board caught my attention.

**Reply by schreibfaul1** [reply]
There are obviously no direct pins for I2S. Perhaps pins have to be configured for this, I don't know.

**Comment by schreibfaul1** [comment]
There are indeed I2S pins, but the Adafruit Board is completely unsuitable, the pins are not accessible.
![image](https://github.com/user-attachments/assets/273af38f-4479-4c80-8e6d-d23dda7666f4)

**Comment by al-davis** [comment]
Better spec here :
https://www.mouser.com/ProductDetail/Skyworks-Solutions-Inc/SI4713-B30-GMR?qs=DPMK3kMt%252Bod71xyZpAOAPw%3D%3D
page 17
DCLK pin 17
DIN  pin 13
DFS  pin 14

pg 23   5.4. Digital Audio Interface  & fig 16 pg 24

**Comment by al-davis** [comment]
Well, they sure aren't brought out and it looks like some surgery would be
involved.

**Comment by al-davis** [comment]
Found another development board that uses the SI4713

https://www.digikey.com/en/products/detail/mikroelektronika/MIKROE-2822/8024052

It also does not bring out the I2S pins but it does support SPI.  Make me
a little concerned that I have not found where that function has been use
on this chip.

**Comment by schreibfaul1** [comment]
That doesn't look usable. Maybe the only option is to develop a board yourself, you need:

for I2S: DIN, GPO3DCLK, DFS
for I2C: SDA, SCL
then TXO, RFGND for the antenna
and VDD, GND and RST

**Comment by al-davis** [comment]
After viewing this video https://www.youtube.com/watch?v=i8re0nc_FdY  I've decided to abandon the SI4713 entirely.  Seems there are some very good reasons no one has implemented I2S on this chip.  I'd like to proceed with either a QN8066 of QN8007 but can not find any. Seems like a worthwhile project but in the end I may just continue on with a commercial transmitter.  Thanks you for your input.
